python/PiFinder/state.py

```diff
@@ -11,14 +11,34 @@
 import pytz
 from PiFinder import config
 import logging
+from typing import List
+from PiFinder.composite_object import CompositeObject
+
+
+class Recent:
```

Comment:
How about using `__iter__` and `append` for this and naming it RecentList?  The semantics would then be just like a python list 🙌   You could even sub-class list itself and replace just the `append` method to implement the same uniqueness/ordering your have here


```
class RecentObjectsList(list):
    def append(self, v: CompositeObject):
        try:
              # remove the object if it's already here, so we can add it to the end
              super().remove(v)
         except ValueError:
             pass
         # add it to the end of the list
         super().append(v)
```